vagrant/roles/config.collect/tasks/main.yml

```diff
@@ -0,0 +1,27 @@
+---
+- name: Create config directory structure
+  file:
+    path: "{{ config.statedir }}/{{ inventory_hostname }}/config"
+    state: directory
+  delegate_to: localhost
+
+- name: Copy common playbook configuration
+  copy:
+    src: ansible/config.yml
+    dest: "{{ config.statedir }}"
+  delegate_to: localhost
+  run_once: true
+
+- name: Process backend specific tasks
+  include_role:
+    name: 'sit.{{ config.be.name }}'
+    tasks_from: config/main.yml
```

Comment:
Instead of delegating the task to copy data to the host, maybe it would be cleaner to create a directory in the VM, let the backend and other modules copy configuration data there, and then copy the entire directory to the host in a single step from this role.

Reply:
Updated to have a `configdir` per VM to collect necessary configuration data.